docs/using.md

```diff
@@ -107,11 +107,58 @@ steps:
         value: "world"
 ```
 
+### Resource shared between tasks
+
 Pipeline Tasks are allowed to pass resources from previous tasks via
 `providedBy` field. This feature is implemented using Persistent Volume Claim
 under the hood but however has an implication that tasks cannot have any volume
 mounted under path `/pvc`.
 
+### Outputs
+
+`Task` definition can include inputs and outputs resource declaration. If specific set of resources are only declared in output then a copy of resource to be uploaded or shared for next task is expected to be present under the path `/workspace/output/resource_name/`. 
+
+```yaml
+resources:
+  outputs:
+    name: storage-gcs
+steps:
+  - image: objectuser/run-java-jar #https://hub.docker.com/r/objectuser/run-java-jar/
+    command: [jar]
+    args: ["-cvf", "-o", "/workspace/output/storage-gcs/", "projectname.war", "*"]
+    env:
+    - name: "FOO"
+      value: "world"
+```
+
+**Note**: If task is relying on output resource functionality then they cannot mount anything in file path `/workspace/output`.
+
+If resource is declared in both input and output then input resource, then destination path of input resource is used instead of `/workspace/output/resource_name`. 
+
+In the following example Task `tar-artifact` resource is used both as input and output so input resource is downloaded into directory `customworkspace`(as specified in `targetPath`). Step `untar` extracts tar file into `tar-scratch-space` directory , `edit-tar` adds a new file and last step `tar-it-up` creates new tar file and places in `/workspace/customworkspace/` directory. After execution of task steps, (new) tar file in directory `customworkspace` will be uploaded to the bucket defined in `tar-artifact` resource definition.
```

Comment:
can the docs show the complete path to `customworkspace`, i.e. is it `/customworkspace` or is it `/workspace/customworkspace` or `/workspace/output/customworkspace`?